### PR TITLE
bye bye web3 cruft from network provider

### PR DESCRIPTION
**Pull request by moodysalem**

### CODE REVIEW

**Comment by moodysalem** [comment]
worked like a charm in the uniswap frontend

File: packages/network-connector/src/index.ts [hunk 1]
```diff
@@ -1,47 +1,90 @@
...
+  }
+
+  sendAsync(request: any, callback: (error: any, response: any) => void): void {
+    fetch(this.url, {
+      method: 'POST',
+      body: JSON.stringify(request)
+    })
+      .then(
+        result => {
+          return result.json()
```
Comment (MicahZoltu) [inline packages/network-connector/src/index.ts:41]: Missing `result.status` check here.  I believe calling `result.json()` on a not-OK response is likely to fail, and if it does succeed you won't get an object shaped like you expect.

File: packages/network-connector/src/index.ts [hunk 2]
```diff
@@ -1,47 +1,90 @@
...
+        json => {
+          callback(undefined, json)
+        }
+      )
+      .catch(
+        error => {
+          callback(error, undefined)
+        }
+      )
+  }
```
Comment (MicahZoltu) [inline packages/network-connector/src/index.ts:62]: Consider extracting out most of this into an async/await function, adding support for EIP 1193's `request` (future compatible) and improving error handling and checking:
```ts
class MiniRpcProvider {
	public readonly url: string
	public readonly sendAsync = (request: { jsonrpc: '2.0', id: number | string | null, method: string, params?: unknown[] | object}, callback: (error: any, response: any) => void): void => {
		this.request(request.method, request.params)
			.then(result => callback(null, { jsonrpc: '2.0', id: request.id, result }))
			.catch(error => callback(error, null))
	}

	public readonly request = async (method: string, params?: unknown[] | object): Promise<unknown> => {
		const response = await fetch(this.url, {
			method: 'POST',
			body: JSON.stringify({
				jsonrpc: '2.0',
				id: 1,
				method,
				params,
			}),
		})
		if (!response.ok) throw new RequestError(`${response.status}: ${response.statusText}`, -32000)
		const body = await response.json()
		if ('error' in body) {
			// TODO: validate that body.error.message and body.error.code exist
			throw new RequestError(body.error.message, body.error.code, body.error.data)
		} else if ('result' in body) {
			return body.result
		} else {
			throw new RequestError(`Received unexpected JSON-RPC response to ${method} request.`, -32000, body)
		}
	}
}

class RequestError extends Error {
	constructor(message: string, public code: number, public data?: unknown) {
		super(message)
	}
}
```

Comment (MicahZoltu) [reply packages/network-connector/src/index.ts:62]: Should validate whether `sendAsync` is expecting the error in its callback to be a JS `Error` or a JSON-RPC error object.  This code assumes a JS `Error`, but if it is expecting a JSON-RPC error then you'll need to convert.

**Pull request merged**